e2e/e2e_test.go

```diff
@@ -0,0 +1,163 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const blockingPodName = "api-blocker-pod"
+
+var _ = Describe("e2e", func() {
+	var targetNode string
+
+	BeforeEach(func() {
+		// randomly pick a host (or let the scheduler do it by running the blocking pod)
+		// block the api port to make it go Ready false
+		node, err := makeNodeUnready(time.Minute * 5)
+		Expect(err).NotTo(HaveOccurred())
+		targetNode = node
+	})
+
+	AfterEach(func() {
+		removeAPIBlockingPod()
+	})
+	When("Node conditions meets the unhealthy criteria", func() {
+		It("Remediates a host", func() {
+			Expect(nhcRemediatesTheHost(targetNode, time.Minute*1)).To(Succeed())
+			Expect(nodeObjectIsRemoved(targetNode, time.Minute*1)).To(Succeed())
+		})
+	})
+})
+
+func nodeObjectIsRemoved(nodeName string, timeout time.Duration) error {
+	fmt.Fprintf(GinkgoWriter, "Node %v should be removed by ppill. Waiting for %v\n", nodeName, timeout)
+	err := wait.Poll(time.Second*5, timeout, func() (done bool, err error) {
+		_, err = clientSet.CoreV1().Nodes().
+			Get(context.Background(),
+				nodeName,
+				metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				fmt.Fprintf(GinkgoWriter, "Node %v is not fount and probably was removed by ppill\n", nodeName)
+				return true, nil
+			}
+			return false, err
+		}
+		return true, fmt.Errorf("Node %v still found and was not removed", nodeName)
```

Comment:
hmm doesn't this mean that we will return after the first attempt?
to my understanding `Poll()` returns when true being returned

Reply:
yeah that should be a false